src/main/java/com/toptrumps/core/engine/GameEngine.java

```diff
@@ -16,116 +15,180 @@
+     *                 which will compose the game play
      */
     public GameEngine(String deckFile) {
+        // Object responsible for shuffling and splitting the deck
         this.dealer = new Dealer(deckFile);
     }
 
-    private ArrayList<Player> createPlayers(int numberOfOpponents) {
-        List<AIPlayer> aiPlayers = createAIPlayers(numberOfOpponents);
-        return new ArrayList<Player>() {{
-            add(new Player(0, DEFAULT_USER_NAME));
-            addAll(aiPlayers);
-        }};
-    }
-
-    private List<AIPlayer> createAIPlayers(int numberOfOpponents) {
-        List<AIPlayer> aiPlayers = new ArrayList<>();
-        for (int i = 1; i <= numberOfOpponents; i++) {
-            AIPlayer aiPlayer = new AIPlayer(i);
-            aiPlayers.add(aiPlayer);
-        }
-        return aiPlayers;
-    }
-
+    /**
+     * Populates the given players with shuffled decks.
+     *
+     * @param players player list to whom the card decks must be assigned
+     */
     public void assignDecks(ArrayList<Player> players) {
         final int numberOfPlayers = players.size();
+        // Gets the shuffled and split decks from the dealer
         final ArrayList<ArrayList<Card>> decks = dealer.dealCards(numberOfPlayers);
-
-        // Distribute the split decks to the game players
+        // Distributes the split decks to the game players
         for (int i = 0; i < numberOfPlayers; i++) {
             players.get(i).setDeck(decks.get(i));
         }
     }
 
+    /**
+     * Randomly assigns a player from the given array as the
+     * active player of the game.
+     *
+     * @param players player list used to pick the active player
+     * @return the player which has been randomly marked as active
+     */
     public Player assignActivePlayer(List<Player> players) {
         final int numberOfPlayers = players.size();
+        // Generates a random integer that corresponds to a valid index
         int randomIndex = RandomGenerator.getInteger(0, numberOfPlayers - 1);
+        // Selects a player from the given collection and marks it as "active"
         Player activePlayer = players.get(randomIndex);
         activePlayer.setActive(true);
         return activePlayer;
     }
 
+    /**
+     * Creates the players of this current game based on the numberOfPlayers
+     * provided. By convention, the first player is always the human player.
+     * In sequence, it distributes shuffled decks between all players.
+     *
+     * @param numberOfOpponents the number of AI opponents that must be created
+     * @return a list with all the initialized game players
+     */
     public ArrayList<Player> startUp(int numberOfOpponents) {
+        // Creates players
         ArrayList<Player> players = createPlayers(numberOfOpponents);
+        // Assigns shuffled decks
         assignDecks(players);
         return players;
     }
 
+    /**
+     * Compares each player's selected attribute and return players who possessed
```

Comment:
returns - I'm being picky but I think it needs the s on the end ;)

Reply:
Thank you!!! :)